hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -875,6 +916,86 @@ ChangesetInfoPtr XmlChangeset::splitChangeset(ChangesetInfoPtr changeset)
     updateFailedChangeset(changeset);
     return split;
   }
+  //  Try to use the split hint to split out one failing element
+  if (splitHint != "")
+  {
+    long member_id = 0;
+    long element_id = 0;
+    ElementType::Type member_type = ElementType::Unknown;
+    ElementType::Type element_type = ElementType::Unknown;
+    //  See if the hint is something like: Placeholder node not found for reference -145213 in way -5687
+    if (matchesPlaceholderFailure(splitHint, member_id, member_type, element_id, element_type))
+    {
+      //  Use the type and id to split the changeset
+      if (element_type == ElementType::Way)
+      {
+        for (int current_type = ChangesetType::TypeCreate; current_type != ChangesetType::TypeMax; ++current_type)
+        {
+          if (changeset->contains(element_type, (ChangesetType)current_type, element_id))
+          {
+            XmlWay* way = dynamic_cast<XmlWay*>(_allWays[element_id].get());
+            //  Add the way to the split and remove from the changeset
+            split->add(element_type, (ChangesetType)current_type, way->id());
+            changeset->remove(element_type, (ChangesetType)current_type, way->id());
+            return split;
```

Comment:
The `Placeholder` node/way/relation fails when a sub element doesn't exist.  That means that the element cannot be added/modified because of it.  So this code splits off just that one element into the split changeset.  A changeset of one element that fails is then moved to the failed state and we don't have to bisect a changeset of 1000+ elements over and over and over looking for that one element.